adafruit_gps.py

```diff
@@ -125,12 +125,22 @@ def _read_degrees(data: List[float], index: int, neg: str) -> float:
     return x
 
 
-def _read_int_degrees(data: List[float], index: int, neg: str) -> Tuple[int, float]:
-    deg = data[index] // 1000000
-    minutes = data[index] % 1000000 / 10000
+def _read_deg_mins(data: List[str], index: int, neg: str) -> Tuple[int, float]:
+    # the degrees come in different formats and vary between latitudes and
+    # longitudes, which makes parsing tricky:
+    # for latitudes: ddmm,mmmm (0 - 7 decimal places, not zero padded)
+    # for longitudes: dddmm,mmmm (0 - 7 decimal places, not zero padded)
+    int_part, _, minutes_decimal = data[index].partition(".")
```

Comment:
`partition` will always return `tuple[str, str, str] so it's save to unpack. `split` will return `list[str]` so we cannot safely unpack - this was my though behind this after working a lot with `mypy`...

Reply:
The docs say that `partition()` isn't available on small boards, though I'm not sure if that's up to date.  If that's the case and this library can run on smaller boards, it may be worth using `split()` if there's ever only one period in the string, which seems to be the case.

Reply:
Yeah, since it's disabled for small boards, let's use `split()`

Reply:
mhm, that's a bit annoying but nothing we can do I guess - I found another edge case where partition is needed. It's when the value is provided as an integer - this test-case was testing it `(["00123", "e"], (1, 23), "w")`. That only works with partition.
I had to add some extra code to make this work with `split`